tools/link.py

```diff
@@ -574,25 +574,91 @@ def include_and_export(name):
       settings.EXPORTED_RUNTIME_METHODS += ['ExitStatus']
 
 
+def set_initial_memory():
+  user_specified_initial_heap = 'INITIAL_HEAP' in user_settings
+
+  # INITIAL_HEAP cannot be used when the memory object is created in JS.
```

Comment:
Can you mention why this is not possible?   I guess its because we don't have easy access to the static data size in order to calculate the total memory needed for "stack+heap+data"?